packages/timeline/src/ReactAxisGanttSeries.ts

```diff
@@ -0,0 +1,253 @@
+            width: w,
+            height: contentHeight
+        });
+        this._bottomWA.resize({
+            width: w,
+            height: axisHeight
+        });
+        this.top().render();
+        this.bottom().render();
+        this.center().render();
+    }
+
+    enter(domNode, element){
+        super.enter(domNode, element);
+
+        this._gantt.click = (row, col, sel) => {
+            this.click(row, col, sel);
+        };
+
+        this._gantt.dblclick = (row, col, sel) => {
+            this.dblclick(row, col, sel);
+        };
+
+        this.top(this._topAxis);
+        this.center(this._gantt);
+        this.bottom(this._bottomAxis);
+
+        this.resizeWrappers();
+
+        this._gantt.zoomedHook = (transform) => {
+            this.onzoom(transform);
+        };
+    }
+
+    onzoom(transform) {
+        console.log("called onzoom");
+        const w = this.width();
+        const low = this._gantt._minStart;
+        const high = this._gantt._maxEnd;
+        const range = high - low;
+        const wpp = range / w;
+        const nextLow = Math.floor(low - (wpp * (transform.x / transform.k)));
+        const nextHigh = Math.ceil((range / transform.k) + nextLow);
+
+        this._topAxis
+            .fontFamily(this.axisFontFamily())
+            .fontSize(this.axisFontSize())
+            .tickLength(this.axisTickLength())
+            .low(nextLow)
+            .high(nextHigh)
+            .lazyRender()
+            ;
+        this._bottomAxis
+            .fontFamily(this.axisFontFamily())
+            .fontSize(this.axisFontSize())
+            .tickLength(this.axisTickLength())
+            .low(nextLow)
+            .high(nextHigh)
+            .lazyRender()
+            ;
+    }
+
+    update(domNode, element){
+        super.update(domNode, element);
+        this._topAxis.tickFormat(this.tickFormat()).render();
+        this._bottomAxis.tickFormat(this.tickFormat()).render();
+        this._gantt.render();
+    }
+
+    columns(): string[];
+    columns(_: string[]): this;
+    columns(_?: string[]): this | string[] {
+        const retVal = super.columns.apply(this, arguments);
+        if(arguments.length > 0) {
+            this._gantt.columns(_);
+        }
+        return retVal;
+    }
+
+    data(): IAxisGanttData[];
+    data(_: IAxisGanttData[]): this;
+    data(_?: IAxisGanttData[]): this | IAxisGanttData[] {
+        const retVal = super.data.apply(this, arguments);
+        if(arguments.length > 0) {
+            const ganttData: any[] = this.data().map(n=>{
+                const ret = [...n];
+                ret[1] = isNaN(n[1] as any) ? new Date(n[1]).getTime() : Number(n[1]);
+                ret[2] = isNaN(n[2] as any) ? new Date(n[2]).getTime() : Number(n[2]);
+                return ret;
+            });
+            this._gantt._minStart = Math.min(...ganttData.map(n=>n[1])) ?? 0;
+            this._gantt._maxEnd = Math.max(...ganttData.map(n=>n[2])) ?? 1;
```

Comment:
d3Extent would be better here.

Reply:
I am not sue how big ganttData is, but you going from 6->8 loops over it to 1

Reply:
Also if your min == -100 and your max == 0, your code will change that to:  -100 -> 1

Reply:
I tried to figure out a way to use extent here but it looks like d3.extent's accessor doesn't support individual values for min and max. So I've changed this to use d3.min and d3.max.

Reply:
Yea - you would need a common accessor.